### PR TITLE
PERF-4889 Remove the 'numInitialChunks' parameter used with 'ShardCollection' command

### PR DESCRIPTION
**Pull request by robsndr**
**Jira Ticket:** [PERF-4889](https://jira.mongodb.org/browse/PERF-4889)

**Whats Changed:**  
The numInitialChunks parameter was deprecated in [SERVER-74747](https://jira.mongodb.org/browse/SERVER-74747), and now 1 chunk per shard is created by default. 
Adjust workloads accordingly. 

**Patch testing results:**  
Example patch run: https://spruce.mongodb.com/version/655f8149e3c3312d2f4f6616/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: src/workloads/query/LookupColocatedData.yml
Comment (pierlauro) [inline src/workloads/query/LookupColocatedData.yml:152]: Pointing out for other reviewers the chose around this shard key change: since `numInitialChunks` was being ignored, the chunks were actually created on all shards so the distributed lookup was faster (because it was not happening on just one node) and performances improved.
Changing the shard key to non-hashed actually forces the collection to be created on one node and brings performances back to "normal"

**Review by pierlauro — APPROVED** [review]
LGTM

File: src/workloads/query/LookupColocatedData.yml
Comment (jimoleary) [reply src/workloads/query/LookupColocatedData.yml:152]: Thanks! I was wondering why the numInitialChunks params were being deleted.

File: src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml
Comment (jimoleary) [inline src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml:41]: Will older mongo versions still have a default of 2 chunks per shard and will that be significant?

**Review by jimoleary — COMMENTED** [review]
The changes look good to me (I'll defer to your superior knowledge) but I have a couple of questions:

1. These tests run on older versions (most 6 and up, some 5 and up) , will these changes work as expected on them?
2. Is this change likely to cause a change in performance? If yes then you should let the perf buildbarons know before this PR is merged.

File: src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml
Comment (robsndr) [reply src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml:41]: If the updated workload is ran against an older version, the default of 2 chunks per shard will created. It seems that using the default value should not affect the performance in this case.

Comment (jimoleary) [reply src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml:41]: Can you confirm that with a quick patch?

Comment (robsndr) [reply src/workloads/query/TenMDocCollection_ObjectId_Sharded.yml:41]: The performance of those workloads shall be the same, because data will still be evenly distributed across all shards, regardless of the number of chunks used.  

A patch using 6.0 is running [here](https://spruce.mongodb.com/version/6560a5632a60ed45b4d9ec1f/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

**Review by jimoleary — APPROVED** [review]
lgtm

There are some perf differences in the sharding commands that can be ignored, maybe some of the other changes will create change points / BFs but we can probably deal with them afterwards if this change needs to be merged anyway.

Maybe let the buildbarons know.

**Comment by robsndr** [comment]
evergreen merge

**Pull request merged**
